src/engine/math_util.c

```diff
@@ -53,6 +58,17 @@ u32 random_u16(void) {
     return gRandomSeed16;
 }
 
+// Generate a pseudorandom integer from -32768 to 32767.
+s16 random_s16(void) {
+    u16 randomUnsignedValue = random_u16();
+    return *((s16*) &randomUnsignedValue);
```

Comment:
I want to say this will just work if you just do `return (s16)random_u16()` (and it will save you from directly addressing stack memory)

Reply:
I don't think this function is necessary at all tbh. It's not exactly difficult to add a manual s16 cast as needed, and you don't need negative numbers most of the time.

Reply:
> I want to say this will just work if you just do `return (s16)random_u16()` (and it will save you from directly addressing stack memory)

Oh ok that makes total sense, thank you

Reply:
> I don't think this function is necessary at all tbh.

Fair enough.